vault/router.go

```diff
@@ -91,6 +91,15 @@ func (r *Router) Remount(src, dst string) error {
 
 	// Update the mount point
 	r.root.Delete(src)
+	mountEntry, ok := raw.(*mountEntry)
```

Comment:
We could, although it doesn't buy us much -- we'd still have to ensure that it is a dynamicSystemView, and if so, invoke remount(newpath) vs dynSysView.path = newpath.